actionview/lib/action_view/template/handlers/erb.rb

```diff
@@ -58,6 +58,7 @@ def add_expr_escaped(src, code)
         def add_stmt(src, code)
           flush_newline_if_pending(src)
           super
+          flush_newline_if_comment(src, code)
```

Comment:
Thanks for pointing this out guys. I thought about it, and the benchmark didn't alarm me:


Running a 1000 lines erb file containing this repeated code
```erb
<% if true %>
  <% [1, 2, 3, 4].each do |number| %>
    <li> <%= number %>
  <% end %>
  <% if false %>
    <div>Test</div>
  <% end %>
  <% if true %>
    <% [5, 6, 7, 8].each do |number2| %>
      <p> <%= number2 %> </p>
    <% end %>
    <% ['John', 'Cena'].each do |name| %>
      <p> <%= name %> </p>
    <% end %>
  <% end %>
<% end %>
```

5000 iterations on [Erubis#src](https://github.com/rails/rails/blob/master/actionview/lib/action_view/template/handlers/erb.rb#L123)
```
                 user        system    total     real
without_regex    36.110000   0.240000  36.350000 ( 36.359235)
with_regex       39.140000   0.280000  39.420000 ( 39.467811)
```

I'm not very used to do benchmarking, sorry if the way I did it is wrong. Thanks!